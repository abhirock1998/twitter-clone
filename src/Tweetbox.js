import { Button, Avatar } from "@material-ui/core";
import React, { useState } from "react";
import "./tweetbox.css";
import db from "./firebase";
import firebase from "firebase";
function Tweetbox({user}) {
  const [tweetMessage, setTweetMessage] = useState("");
  const [tweetImage, setTweetImage] = useState("");
  const randomString = () => {
    var randChara =
      "abcdefghijklmnopqrstuvwxyxABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
    var result = "";
    for (var i = 0; i < 20; i++) {
      result += randChara[Math.floor(Math.random() * randChara.length)];
    }

    return result;
  };
  const disabled = tweetMessage === "" ? true : false;
  const sendTweet = (e) => {
    e.preventDefault();
    db.collection("posts").add({
      avatar: user?.photoURL,
      username: user?.displayName,
      name: user?.displayName,
      text: tweetMessage,
      image: tweetImage,
      verified: user?.verified,
      timestamp: firebase.firestore.FieldValue.serverTimestamp(),
      key: randomString(),
    });
    setTweetMessage("");
    setTweetImage("");
  };
  return (
    <div data-testid='tweetBox' className="tweetBox">
      <form>
        <div className="tweetBox_input">
          <Avatar src={user?.photoURL} />
          <input
            value={tweetMessage}
            type="text"
            onChange={(e) => setTweetMessage(e.target.value)}
            placeholder="What's happening?"
          />
        </div>
        <input
          value={tweetImage}
          type="text"
          onChange={(e) => setTweetImage(e.target.value)}
          className="tweetBox_imageInput"
          placeholder="Optional: Enter image URL"
        />
        <Button
          disabled={disabled}
          onClick={sendTweet}
          className="tweetBox_button"
        >
          Tweet
        </Button>
      </form>
    </div>
  );
}

export default Tweetbox;
