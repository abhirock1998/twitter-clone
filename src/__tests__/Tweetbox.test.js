import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import Tweetbox from "../Tweetbox";
import SearchIcon from "@material-ui/icons/Search";

describe("<Tweetbox/> ", () => {
  it("render the <Tweetbox/> with dummy data", () => {
    const {container,getByTestId,queryByTestId} = render(
       <Tweetbox/>
    )
    

    expect(queryByTestId("tweetBox")).toBeTruthy();
    fireEvent.click(getByTestId('tweetBox'));
    expect(queryByTestId('tweetBox')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
   
  });
});
