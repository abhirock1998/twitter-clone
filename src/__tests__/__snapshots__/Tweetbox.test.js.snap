// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tweetbox/>  render the <Tweetbox/> with dummy data 1`] = `
<div
  class="tweetBox"
  data-testid="tweetBox"
>
  <form>
    <div
      class="tweetBox_input"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <input
        placeholder="What's happening?"
        type="text"
        value=""
      />
    </div>
    <input
      class="tweetBox_imageInput"
      placeholder="Optional: Enter image URL"
      type="text"
      value=""
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text tweetBox_button Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Tweet
      </span>
    </button>
  </form>
</div>
`;
